### PR TITLE
Add and update the scripts for the new instrument, kinematic data record & replay

### PR DESCRIPTION
**Pull request by JackHaoyingZhou**
The pull request contributes to the following points:
1. make the scale of everything to be 1:1 (change linear scaling factor from 10 to 1).
2. add kinematic data recording and replaying scripts, correspond the replay scaling factor with the unit conversion information.
3. add an additional argument called "tool_id" to address kinematics and teleoperation scripts for different surgical instrument

Need to do:
1. add new models (include new suturing needle, 3d med phantom, new surgical instrument and etc)
2. I only provide the closed-form FK, but I also have scripts for FK using screw theory, can add to the repository if required
3. the scripts suppose to work with new models, will test after adding the new models.

### CODE REVIEW

**Comment by JackHaoyingZhou** [comment]
Merge FK&IK method into the psmKinematic solver class; create a class variable for the teleoperation scaling factor; in the later future, create a json file to load all tools

**Comment by JackHaoyingZhou** [comment]
Revision done, clean code, ready to merge

File: scripts/surgical_robotics_challenge/psm_arm.py [hunk 1]
```diff
@@ -43,10 +43,10 @@
 #     \version   1.0
 # */
 # //==============================================================================
-from surgical_robotics_challenge.kinematics.psmIK import *
+from surgical_robotics_challenge.kinematics.psmKinematics import *
 from surgical_robotics_challenge.utils.joint_errors_model import JointErrorsModel
 from surgical_robotics_challenge.utils import coordinate_frames
-
+import rospy
```
Comment (adnanmunawar) [inline scripts/surgical_robotics_challenge/psm_arm.py:49]: Do we need rospy? If not, let's remove this import.

File: scripts/surgical_robotics_challenge/kinematics/config/kinematic/psm_400006.json
Comment (adnanmunawar) [inline scripts/surgical_robotics_challenge/kinematics/config/kinematic/psm_400006.json:1]: What is the difference between the files kinematic/psm_*.json and tool/LARGE_NEEDLE*.json

**Review by adnanmunawar — COMMENTED** [review]
This looks great Jack. I just added a few comments for clarification.

File: scripts/surgical_robotics_challenge/psm_arm.py (hunk 1 above)
Comment (JackHaoyingZhou) [reply scripts/surgical_robotics_challenge/psm_arm.py:49]: In line #216-219, rospy was used for the function "_execute_trajectory(...)". Therefore, I believe the rospy import is required.

File: scripts/surgical_robotics_challenge/kinematics/config/kinematic/psm_400006.json
Comment (JackHaoyingZhou) [reply scripts/surgical_robotics_challenge/kinematics/config/kinematic/psm_400006.json:1]: I use the format in dVRK repository. kinematic/psm_xx.json involves the information of the first 3 joint, and the tool/<tool>.json includes the information of the last three joints (total 6 joints).

**Comment by JackHaoyingZhou** [comment]
Wait for asset so that we can do the final merge

**Comment by jabarragann** [comment]
Hi Adnan/Jack,

I went over Jack's PR and had some small comments and possible changes. 

* Currently we copy-pasted the JSON files from the dVRK into the SRC repository, but not all the values from these JSON files are used. Only the DH parameters are read from these files. Should we try to simplify them? That is only keeping the values that are used inside the SRC code base.

* Add a readme on the config folder indicating that 4200006 is an SI large needle driver and that 4000006 is a classic large needle driver. 

* Maybe change the names of kinematic files to include the `AMBF_large_needle_driver.json`. This way we make sure people know these are only for the simulation. Also, specify in the readme which DH parameters are different from the ones in the DVRK repository. 

## Future developments
* Adding to ROS parameters the tool name that is loaded in the simulation so that the FK script can automatically load the correct JSON files. Similar to the way the dVRK loads its kinematic parameters.

**Comment by jabarragann** [comment]
@adnanmunawar and @JackHaoyingZhou a small change before submitting the PR:

1. the teleoperation scripts names are `mtm_psm_pair_teleop_400006.sh` and `mtm_psm_pair_teleop_420006.sh`, but the script to run the environment are `run_env_simple_instruments.sh` and `run_env_real_instruments.sh`.  I would make sure these names consistent.

Probably users won't know that simple instruments are based on the 400006 instrument.

2. Revise all scripts that included FK and IK functionalities. There might be some broken scripts.

**Pull request merged**